tests/lax_numpy_test.py

```diff
@@ -5857,6 +5866,87 @@ def testUfuncInputTypes(self, name, arg_dtypes):
+        _complex_numpy_ufuncs,
+        jtu.dtypes.supported([np.complex64, np.complex128]),
+        ['q1', 'q2', 'q3', 'q4',
+         'pos', 'neg', 'posj', 'negj',
+         'ninf', 'pinf', 'ninfj', 'pinfj',
+         'zero']
+    ))
+  def testUfuncOnComplexPlane(self, name, dtype, region):
+    # TODO(pearu): eliminate all items in the following sets:
+    if (name, region) in {
+        ('absolute', 'q1'), ('absolute', 'q2'), ('absolute', 'q3'), ('absolute', 'q4'),
+        ('exp', 'pos'), ('exp', 'pinfj'), ('exp', 'pinf'), ('exp', 'ninfj'), ('exp', 'ninf'),
+        ('exp2', 'pos'), ('exp2', 'pinfj'), ('exp2', 'pinf'), ('exp2', 'ninfj'), ('exp2', 'ninf'),
+        ('expm1', 'q1'), ('expm1', 'q2'), ('expm1', 'q3'), ('expm1', 'q4'), ('expm1', 'pos'), ('expm1', 'neg'),
+        ('expm1', 'posj'), ('expm1', 'negj'),
+        ('log', 'q1'), ('log', 'q2'), ('log', 'q3'), ('log', 'q4'),
+        ('log1p', 'q1'), ('log1p', 'q2'), ('log1p', 'q3'), ('log1p', 'q4'),
+        ('log1p', 'pos'), ('log1p', 'neg'), ('log1p', 'posj'), ('log1p', 'negj'),
+        ('log1p', 'ninf'), ('log1p', 'ninfj'), ('log1p', 'pinfj'),
+        ('log10', 'q1'), ('log10', 'q2'), ('log10', 'q3'), ('log10', 'q4'), ('log10', 'zero'),
+        ('log10', 'ninf'), ('log10', 'ninfj'), ('log10', 'pinf'), ('log10', 'pinfj'),
+        ('sinh', 'q1'), ('sinh', 'q2'), ('sinh', 'q3'), ('sinh', 'q4'),
+        ('sinh', 'pos'), ('sinh', 'neg'), ('sinh', 'ninf'), ('sinh', 'pinf'),
+        ('cosh', 'pos'), ('cosh', 'neg'), ('cosh', 'ninf'), ('cosh', 'pinf'),
+        ('cosh', 'q1'), ('cosh', 'q2'), ('cosh', 'q3'), ('cosh', 'q4'),
+        ('tan', 'q1'), ('tan', 'q2'), ('tan', 'q3'), ('tan', 'q4'),
+        ('tan', 'negj'), ('tan', 'posj'), ('tan', 'ninf'), ('tan', 'ninfj'), ('tan', 'pinf'), ('tan', 'pinfj'),
+        # warning: numpy square is incorrect on inputs with large absolute value
+        ('square', 'q1'), ('square', 'q2'), ('square', 'q3'), ('square', 'q4'),
+        ('square', 'ninfj'), ('square', 'pinfj'),
+        ('sinc', 'q1'), ('sinc', 'q2'), ('sinc', 'q3'), ('sinc', 'q4'),
+        ('sign', 'q1'), ('sign', 'q2'), ('sign', 'q3'), ('sign', 'q4'),
+        ('sign', 'negj'), ('sign', 'posj'), ('sign', 'ninf'), ('sign', 'ninfj'), ('sign', 'pinf'), ('sign', 'pinfj'),
+    } or (
+      name in {'arcsin', 'arccos', 'arctan', 'arcsinh', 'arccosh', 'arctanh'}
+    ) or (
+      jtu.test_device_matches(["cpu"]) and (name, region) in {
+        ('exp2', 'q1'), ('exp2', 'q4'),
+        ('sinh', 'negj'), ('sinh', 'posj'),
+      }
+    ) or (
+      jtu.test_device_matches(["cpu"]) and dtype == np.complex64 and (name, region) in {
+        ('square', 'negj'), ('square', 'neg'), ('square', 'posj'), ('square', 'pos'),
+      }
+    ):
+      self.skipTest(f"np.{name} & jnp.{name} have differences in the given region.")
+    jnp_op = getattr(jnp, name)
+    np_op = getattr(np, name)
+    rtol = {np.complex64: 1e-6, np.complex128: 1e-14}[dtype]
+
+    with warnings.catch_warnings():
```

Comment:
You should prefer `jtu.ignore_warning()` to ignore specific warnings, rather than silencing all warnings.

Reply:
Fixed.